torch_geometric_temporal/nn/attention/2sagcn.py

```diff
@@ -0,0 +1,307 @@
+
+        self.A = self.get_adjacency_matrix()
+
+    def get_adjacency_matrix(self):
+        A = get_spatial_graph(self.num_nodes, self.self_link, self.inward, self.outward)
+
+        return A
+
+def conv_branch_init(conv, branches):
+    weight = conv.weight
+    n = weight.size(0)
+    k1 = weight.size(1)
+    k2 = weight.size(2)
+    nn.init.normal_(weight, 0, math.sqrt(2. / (n * k1 * k2 * branches)))
+    nn.init.constant_(conv.bias, 0)
+
+
+def conv_init(conv):
+    nn.init.kaiming_normal_(conv.weight, mode='fan_out')
+    nn.init.constant_(conv.bias, 0)
+
+
+def bn_init(bn, scale):
+    nn.init.constant_(bn.weight, scale)
+    nn.init.constant_(bn.bias, 0)
+
+
+class unit_tcn(nn.Module):
+    r"""
+    Temporal Convolutional Block applied to nodes in the Two-Stream Adaptive Graph Convolutional Network as originally implemented in the `Github Repo <https://github.com/lshiwjx/2s-AGCN>`.
+    For implementational details see https://arxiv.org/abs/1805.07694
+    Args:
+        in_channels (int): Number of input features.
+        out_channels (int): Number of output features.
+        kernel_size (int): Convolutional kernel size.
+        stride (int): Convolutional kernel stride.
+    """
+    def __init__(self, in_channels:int, out_channels:int, kernel_size: int=9, stride: int=1):
+        super(unit_tcn, self).__init__()
+        pad = int((kernel_size - 1) / 2)
+        self.conv = nn.Conv2d(in_channels, out_channels, kernel_size=(kernel_size, 1), padding=(pad, 0),
+                              stride=(stride, 1))
+
+        self.bn = nn.BatchNorm2d(out_channels)
+        self.relu = nn.ReLU(inplace=True)
+        conv_init(self.conv)
+        bn_init(self.bn, 1)
+
+    def forward(self, x):
+        x = self.bn(self.conv(x))
+        return x
+
+
+class unit_gcn(nn.Module):
+    r"""
+    Graph Convolutional Block applied to nodes in the Two-Stream Adaptive Graph Convolutional Network as originally implemented in the `Github Repo <https://github.com/lshiwjx/2s-AGCN>`.
+    For implementational details see https://arxiv.org/abs/1805.07694. 
+    Temporal attention, spatial attention and channel-wise attention will be applied. 
+    Args:
+        in_channels (int): Number of input features.
+        out_channels (int): Number of output features.
+        A (Tensor array): Adaptive Graph.
+        coff_embedding (int, optional): Coefficient Embeddings
+        num_subset (int, optional): Subsets for adaptive graphs, see :math:`\mathbf{A}, \mathbf{B}, \mathbf{C}` in https://arxiv.org/abs/1805.07694 for details
+        adaptive (bool, optional): Apply Adaptive Graph Convolutions.
+        attention (bool, optional): Apply Attention.
+    """
+    def __init__(self, in_channels:int, out_channels:int, A:torch.FloatTensor, coff_embedding:int=4, num_subset:int=3, adaptive: bool=True, attention: bool=True):
+        super(unit_gcn, self).__init__()
+        inter_channels = out_channels // coff_embedding
```

Comment:
Please break up the init to submethods. Use private submethods for meaningful subunits of code.